src/Authentication/Authenticators/Session.php

```diff
@@ -149,6 +152,9 @@ public function attempt(array $credentials): Result
         // Update the user's last used date on their password identity.
         $user->touchIdentity($user->getEmailIdentity());
 
+        // Check ID_TYPE_EMAIL_ACTIVATE identity
+        $hasEmailActivate = $this->setAuthActionEmailActivate();
```

Comment:
@kenjis , let's say the user uses his own custom actions, for example `sms_activate` , `sms_2fa` , have you considered this?
Or am I wrong?

Reply:
This PR only fixes the bugs. Refactoring will be done in the next PR #397.